types/node/v4/index.d.ts

```diff
@@ -558,13 +558,17 @@ declare module "http" {
     import * as stream from "stream";
 
     // incoming headers will never contain number
-    export interface IncomingHttpHeaders {
-        [header: string]: string | string[];
+    export type IncomingHttpHeaders = {
+        [header: string]: string
+    } & {
+        'set-cookie'?: string[]
     }
 
     // outgoing headers allows numbers (as they are converted internally to strings)
-    export interface OutgoingHttpHeaders {
-        [header: string]: number | string | string[] | undefined;
+    export type OutgoingHttpHeaders = {
```

Comment:
All outgoing headers can be specified as array by user, not just set-cookie. This differs from incoming headers where node combines them during parsing.